corehq/ex-submodules/dimagi/utils/django/email.py

```diff
@@ -52,15 +52,15 @@ def mark_local_bounced_email(bounced_addresses, message_id):
             )
 
 
-def get_valid_recipients(recipients, domain=None):
+def get_valid_recipients(recipients, domain=None, is_conditional_alert=False):
     """
     This filters out any emails that have reported hard bounces or complaints to
     Amazon SES
     :param recipients: list of recipient emails
     :return: list of recipient emails not marked as bounced
     """
     from corehq.toggles import BLOCKED_DOMAIN_EMAIL_SENDERS
-    if domain and BLOCKED_DOMAIN_EMAIL_SENDERS.enabled(domain):
+    if domain and BLOCKED_DOMAIN_EMAIL_SENDERS.enabled(domain) and is_conditional_alert:
```

Comment:
I would put is_conditional_alert first since it does not need any computation and can short circuit this if statement faster.

Reply:
I like it https://github.com/dimagi/commcare-hq/pull/35511/commits/04eb213e092b51c4843695ff5d2718096088d157